src/app/tabcontent/tabcontent.component.ts

```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-dashboard-tabcontent',
+  templateUrl: './tabcontent.component.html',
+  styleUrls: ['./tabcontent.component.css']
+})
+export class TabcontentComponent implements OnInit {
+  @Input() columns = 1;
+  @Input() rows = 1;
+  @Input() chars = [];
```

Comment:
What is `columns`, `rows`, `chars`? Please remove them. We will add later when it is necessary.